implementations/rust/ockam/ockam_node/src/context/receive_message.rs

```diff
@@ -1,64 +1,20 @@
 use core::sync::atomic::Ordering;
-use core::time::Duration;
 
-use ockam_core::{Message, RelayMessage, Result, Routed};
+use ockam_core::compat::sync::Arc;
+use ockam_core::{IncomingAccessControl, Message, RelayMessage, Result, Routed};
 
-use crate::debugger;
+use crate::context::message_options::MessageWait;
 use crate::error::*;
 use crate::tokio::time::timeout;
-use crate::{Context, DEFAULT_TIMEOUT};
-
-pub(super) enum MessageWait {
-    Timeout(Duration),
-    Blocking,
-}
-
-/// Full set of options to `send_and_receive_extended` function
-pub struct MessageReceiveOptions {
-    message_wait: MessageWait,
-}
-
-impl Default for MessageReceiveOptions {
-    fn default() -> Self {
-        Self::new()
-    }
-}
-
-impl MessageReceiveOptions {
-    /// Default options with [`DEFAULT_TIMEOUT`]
-    pub fn new() -> Self {
-        Self {
-            message_wait: MessageWait::Timeout(DEFAULT_TIMEOUT),
-        }
-    }
-
-    pub(super) fn with_message_wait(mut self, message_wait: MessageWait) -> Self {
-        self.message_wait = message_wait;
-        self
-    }
-
-    /// Set custom timeout
-    pub fn with_timeout(mut self, timeout: Duration) -> Self {
-        self.message_wait = MessageWait::Timeout(timeout);
-        self
-    }
-
-    /// Set custom timeout in seconds
-    pub fn with_timeout_secs(mut self, timeout: u64) -> Self {
-        self.message_wait = MessageWait::Timeout(Duration::from_secs(timeout));
-        self
-    }
-
-    /// Wait for the message forever
-    pub fn without_timeout(mut self) -> Self {
-        self.message_wait = MessageWait::Blocking;
-        self
-    }
-}
+use crate::Context;
+use crate::{debugger, MessageReceiveOptions};
 
 impl Context {
     /// Wait for the next message from the mailbox
-    pub(crate) async fn receiver_next(&mut self) -> Result<Option<RelayMessage>> {
+    pub(crate) async fn receiver_next(
```

Comment:
Why isn't that called `receive_next`?

Reply:
Maybe because it pulls the next message from the receiver. But was called like this from the very beginning)